tests/test_resolver.py

```diff
@@ -139,6 +151,70 @@ def test_resolve_extra_wheel():
     for wheel in (project1_wheel, project2_wheel):
       safe_copy(wheel, os.path.join(td, os.path.basename(wheel)))
 
-    resolved_dists = do_resolve_multi(['project1[foo]'], find_links=[td])
+    resolved_dists = local_resolve_multi(['project1[foo]'], find_links=[td])
     assert ({_parse_requirement(req) for req in ('project1==1.0.0', 'project2==2.0.0')} ==
             {_parse_requirement(resolved_dist.requirement) for resolved_dist in resolved_dists})
+
+
+def resolve_wheel_names(**kwargs):
+  return [
+    os.path.basename(resolved_distribution.distribution.location)
+    for resolved_distribution in resolve_multi(
+      requirements=['p537==1.0.4'],
+      transitive=False,
+      **kwargs
+    )
+  ]
+
+
+@pytest.fixture(scope="module")
+def p537_resolve_cache():
+  return safe_mkdtemp()
+
+
+@pytest.mark.skipif(PY_VER < (3, 5) or IS_PYPY,
+                    reason="The p537 distribution only builds for CPython 3.5+")
+def test_resolve_current_platform(p537_resolve_cache):
+  resolve_current = functools.partial(resolve_wheel_names,
+                                      cache=p537_resolve_cache,
+                                      platforms=['current'])
+
+  other_python_version = PY36 if PY_VER == (3, 5) else PY35
+  other_python = PythonInterpreter.from_binary(ensure_python_interpreter(other_python_version))
+  current_python = PythonInterpreter.get()
+
+  resolved_other = resolve_current(interpreters=[other_python])
+  resolved_current = resolve_current()
+
+  assert 1 == len(resolved_other)
+  assert 1 == len(resolved_current)
+  assert resolved_other != resolved_current
+  assert resolved_current == resolve_current(interpreters=[current_python])
+  assert resolved_current == resolve_current(interpreters=[current_python, current_python])
```

Comment:
I don't understand this line. When would you ever have two instances of the same interpreter?

Reply:
You wouldn't, but primarily this is test of de-duping and at the unit level, interpreters are handed in as a list, so an API user could hand in dups. More towards reality, given an interpreter constraint (that whole concept has been broken from the get-go really), you might collect a python 2.7.15 - say from pyenv, and a 2.7.17, say from the system. Since wheels / platform specifiers resolve only down to major/minor, these two different interpreters would produce the same wheels in which case we'd want to dedup.